app/Helpers/database/site-award.php

```diff
@@ -172,12 +174,12 @@ function HasPatreonBadge(string $username): bool
     return mysqli_num_rows($dbResult) > 0;
 }
 
-function SetPatreonSupporter(string $username, bool $enable): void
+function SetPatreonSupporter(string $username, int $userId, bool $enable): void
```

Comment:
Since you're already changing these signatures, how difficult would it be to pass a `User` object in?

Reply:
Good point - not difficult at all. Addressed in https://github.com/RetroAchievements/RAWeb/pull/2299/commits/5e2b5b0a0fbb5f1765be19dec59565f8d653e0d5.

Now, a `User` entity is sent all the way down to `AddSiteAward()`.